src/java/org/apache/cassandra/service/reads/ReadTrackingTransformation.java

```diff
@@ -69,6 +69,24 @@ protected Row applyToRow(Row row)
         return super.applyToRow(row);
     }
 
+    @Override
+    protected Row applyToStatic(Row row)
+    {
+        try
+        {
+            if (!row.isEmpty())
+            {
+                readTracker.onRow(row);
```

Comment:
It is arguable if we want to treat static row as a regular row (i.e. eventually increase read row count and not e.g. only the number of bytes read), risking overcounting (from a certain perspective) rows by a factor up to 2.

However, this discussion has likely been resolved in the past as during writes, we indeed do count static rows as regular rows (i.e. bump both the number of rows and the number of bytes written).

If this is not correct, I kindly ask dear reviewers to weigh in.

Reply:
> , risking overcounting (from a certain perspective) rows by a factor up to 2.

What do you mean?  Rows with static data in them should be treated the same as regular rows in terms of counting the bytes and number of rows returned, because static data is not minimized in the result set returned to the user, it is expanded out to each row.

Reply:
Suppose there is only 1 row per partition. 
If the schema says some columns are static we'll always have 2 rows read per partition. If these columns are not static there will be 1 row per partition. The user will always read 1 row (in result set sense) from a partition.
From the user's perspective, the difference in the number of rows read may be seen as twofold. I'm not saying that we do the wrong thing, but that it is a point that was likely discussed in the past.

I don't understand the argument about the expansion of static data out to each row in the result set.
We count static row data size only once, not once per regular row. I'd expect to count static row data size per each row, and perhaps not count static row towards the row count at all (except in edge cases with no regular rows etc.) if we wanted to reflect static row expansion in the billing numbers.

Reply:
> Suppose there is only 1 row per partition.
If the schema says some columns are static we'll always have 2 rows read per partition. If these columns are not static there will be 1 row per partition. The user will always read 1 row (in result set sense) from a partition.

If we are calling that two rows, then we are doing the wrong thing with counting rows.  We only read one row, not two.  I guess that doesn't really matter in the scheme of things since we don't bill based on rows read, but on bytes.

> We count static row data size only once, not once per regular row

Thinking about this more, our main goal is charging based on work done, so just charging for the static data once makes sense from this perspective.

Reply:
> If we are calling that two rows, then we are doing the wrong thing with counting rows. We only read one row, not two.
Well, that's arguable (and that was my original point).
Two makes sense because regular and static row data are not localized. I.e. we do really need to read two rows. I never thought of that as an implementation detail (but maybe that's wrong?). We are explicit about what static columns are; they are marked in schema etc. - and to compose the final row, we do need to read two rows with disjoint sets of columns and merge them.

Reply:
Ok. You are right. This is how master works. Let’s match that and we can discuss changes to how it works with other later if needed.